src/api/app/controllers/source_project_controller.rb

```diff
@@ -52,15 +53,14 @@ def render_project_packages
 
   # DELETE /source/:project
   def delete
-    project = Project.get_by_name(params[:project])
-
-    # checks
-    unless project.is_a?(Project) && User.current.can_modify?(project)
-      logger.debug "No permission to delete project #{project}"
-      render_error status: 403, errorcode: 'delete_project_no_permission',
-                   message: "Permission denied (delete project #{project})"
-      return
+    begin
+      project = Project.get_by_name(params[:project])
+    rescue UnknownObjectError, ReadAccessError
+      raise Pundit::NotAuthorizedError
```

Comment:
I don't like this at all.... hm.

Reply:
that's just to avoid the to change the api behavior. For sure I can do:

```
rescue_from Pundit::NotAuthorizedError, UnknownObjectError, ReadAccessError with: :handle_pundit_not_authorized
```

is that better?